view/frontend/web/js/view/payment/method-renderer/omise-offsite-installment-method.js

```diff
@@ -56,13 +56,110 @@ define(
                 return this;
             },
 
+            /**
+             * Get Installment minimum
+             * this function respects info from: https://www.omise.co/installment-payment
+             *
+             * NOTE: in the future this function should return data from capabilities object.
+             *
+             * @return {integer}
+             */
+            getInstallmentMinimum(id) {
+                switch (id) {
```

Comment:
An array or an object solution would be more efficient? Don't see much point to use Switch condition here.
(IIRC, saw @jonrandy commented once?)

Reply:
yeah.. maybe, here is more readable IMHO.

Reply:
@guzzilar did you test it previous `grand_total` comment? we force THB, no THB installment will be not selectable. But might be some my mistake here..

Reply:
Readable, probably. But feel a bit off when it comes to code intention.
If you want to remain its readability then why not just:
```js
return {
  'kbank': 500,
  'bbl': 500,
  'bay': 300,
  'first_choice': 300,
  'ktc': 300
}[id] || 0;
```
Or could even be
```js
const minAllowedAmount = {
  'kbank': 500,
  'bbl': 500,
  'bay': 300,
  'first_choice': 300,
  'ktc': 300
}

return minAllowedAmount[id] || 0;
```
(and yes, I agreed on having a fallback condition)

--
A little add, I think `id` is a bit ambiguous in this case... would prefer `bank_code`, a little long but more meaningful.
```js
return minAllowedAmount[bank_code] || 0;
```
Anyway, just personal preference, I'll leave it to you.

Reply:
I prefer the object based solution too

Reply:
Also feels wrong to have these hardcoded in JS. In a config file would be better?

Reply:
no, as heder of function says, it will come from capabilities object, it is all available here..